lib/whitehall/featured_document_updater.rb

```diff
@@ -0,0 +1,44 @@
+module Whitehall
+  class FeaturedDocumentUpdater
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      update_all_features(document)
+    end
+
+  private
+
+    def update_all_features(document)
+      features = Feature.where(document_id: document.id)
+      latest_published_edition = document.editions.latest_published_edition.first
+      return unless latest_published_edition
+
+      public_url = Whitehall.url_maker.public_document_url(latest_published_edition)
+
+      return unless valid_link_type?(latest_published_edition)
+
+      features.each do |feature|
+        offsite_link = OffsiteLink.create(title: latest_published_edition.title,
+                                          summary: latest_published_edition.summary,
+                                          link_type: "content_publisher_#{link_type(latest_published_edition)}",
+                                          url: public_url,
+                                          date: Time.zone.now,
+                                          parent_type: feature.feature_list.featurable_type,
+                                          parent_id: feature.feature_list.featurable_id)
```

Comment:
As this is a polymorphic association it might be easier to read as parent: `feature.feature_list.featurable` though this could come at expense of an additional DB query.

Reply:
Hmmm. Not sure, I'll take a look.